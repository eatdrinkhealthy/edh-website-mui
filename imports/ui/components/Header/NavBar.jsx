import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import { withStyles } from "@material-ui/core/styles";

const pathAndHash = (match, location, toHash) => {
  if (!match) {
    return false;
  }

  return match.url === location.pathname && toHash === location.hash;
};

const styles = theme => ({
  container: {
    // margin: "5px 0 5px 0",
  },
  selected: {
    color: theme.palette.primary.main,
    "&:hover": {
      color: theme.palette.primary.main,
    },
  },
});

// TODO: to set an anchor link active, could check location.hash
// if equal, set Contact to active
// Note, may need to use Link instead of NavLink to do so
const NavBar = ({ classes: { container, selected } }) => (
  <nav className={container}>
    <ul>
      <li>
        <NavLink
          exact
          to="/"
          activeClassName={selected}
          isActive={(m, l) => pathAndHash(m, l, "")}
        >
          HOME
        </NavLink>
      </li>
      <li>
        <NavLink
          exact
          to={{ pathname: "/", hash: "#contact" }}
          activeClassName={selected}
          isActive={(m, l) => pathAndHash(m, l, "#contact")}
        >
          CONTACT
        </NavLink>
      </li>
      <li>
        <NavLink
          exact
          to="/privacy-policy"
          activeClassName={selected}
          isActive={(m, l) => pathAndHash(m, l, "")}
        >
          PRIVACY
        </NavLink>
      </li>
    </ul>
  </nav>
);

NavBar.propTypes = {
  classes: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default withStyles(styles)(NavBar);
