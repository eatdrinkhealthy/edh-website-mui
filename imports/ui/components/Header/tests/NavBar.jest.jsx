/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../NavBar";

describe("<NavBar />", () => {
  it("should match render snapshot", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
