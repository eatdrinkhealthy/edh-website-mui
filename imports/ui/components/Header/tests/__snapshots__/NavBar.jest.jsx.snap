// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match render snapshot 1`] = `
<nav
  className="NavBar-container-1"
>
  <ul>
    <li>
      <a
        aria-current="page"
        className="NavBar-selected-2"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        HOME
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/#contact"
        onClick={[Function]}
      >
        CONTACT
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/privacy-policy"
        onClick={[Function]}
      >
        PRIVACY
      </a>
    </li>
  </ul>
</nav>
`;
